src/odemis/acq/stream/_sync.py

```diff
@@ -1054,6 +1006,193 @@ def _runAcquisition(self, future):
             self._current_future = None
             self._acq_done.set()
 
+    def _startLeeches(self, rep_time, rep, tot_num):
+        """
+        A leech can be drift correction (dc) and/or probe/sample current (pca).
+        During the leech time the drift correction and/or the probe current measurements are conducted.
+        Start a counter np (= next pixel) until a leech is called.
+        Leech is called after a well specified number of acquired pixel positions.
+        :param rep_time (0<float): dwell time of one pixel
+        :param rep (int, int): 2D grid of pixel positions to be acquired
+        :param tot_num (int): total number of e-beam positions
+        :return:
+            leech_np (list of 0<int or None): for each leech, number of pixels before the leech should be
+                                              executed again. It's automatically updated inside the list.
+                                              (np = next pixels)
+            leech_time_ppx (float): extra time needed on average for a single pixel for all leches (s)
+        """
+
+        leech_np = []
+        leech_time = 0  # how much time leeches will cost
+        for l in self.leeches:
+            # TODO: store each np, as an approximation of how often it should
```

Comment:
You can drop  this TODO, as actually, that's precisely what `leech_time_ppx` does.